test/e2e/auth/node_authn.go

```diff
@@ -97,7 +97,7 @@ func createNodeAuthTestPod(f *framework.Framework) *v1.Pod {
 		Spec: v1.PodSpec{
 			Containers: []v1.Container{{
 				Name:    "test-node-authn",
-				Image:   imageutils.GetE2EImage(imageutils.Hostexec),
+				Image:   imageutils.GetE2EImage(imageutils.Agnhost),
```

Comment:
non-blocking nit: I know the default for the Agnhost was to run `pause` but do you think we should make that explicit via the args here?

Reply:
we could. It's probably safer. Then we'll have to do the same everywhere.